Project1/calculator.cpp

```diff
@@ -0,0 +1,35 @@
+#include "calculator.h"
+
+double Calculator::add(double x, double y) {
+	return x + y;
+}
+
+double Calculator::substruct(double x, double y) {
+	return x - y;
+}
+
+double Calculator::multiply(double x, double y) {
+	return x * y;
+}
+
+double Calculator::divide(double x, double y) {
+	if (y == 0.0) {
+		throw CalculatorException::DIVIDE_BY_ZERO;
```

Comment:
What happens if I write the following code:

try {
    calc.divide(5, 0);
catch (const std::exception e) {
      // handle error
}

Will I be able to handle the error?
Our exceptions should be hierarchal to allow catching a more specific exception.
Specifically, we should always have all our exceptions inherit from std::exception, since that's the standard "BaseException" in modern c++.